ghcide/src/Development/IDE/GHC/Compat.hs

```diff
@@ -5,212 +5,164 @@
 {-# LANGUAGE ConstraintKinds   #-}
```

Comment:
This module can moved entirely to `Core.hs`. Probably sensible

Reply:
Would it be worth to keep it in order to reexport everything

Reply:
I figured, the purpose is roughly this:
`Core.hs` handles simple changes in the GHC API (e.g. renames of functions). Other Compat modules may import `Core.hs`, to avoid the amount of CPP statements.
`Compat.hs` itself, depends on 'Core.hs' and on the most common modules from the `Compat/*` hierarchy. As such, it can provide backwards compatible functions that depend in some way on modules from the `Compat/*` hierarchy. Practical example:
```haskell
runPp :: Logger -> DynFlags -> [Option] -> IO ()
runPp =
#if MIN_VERSION_ghc(9,2,0)
    SysTools.runPp
#else
    const SysTools.runPp
#endif
```

Older GHC versions don't have a logger type. Thus, we can't have this function in 'Core.hs'. However, since it is basically a simple rename, it is not really necessary to add another module in the  'Compat/*' hierarchy. Thus, this function is defined in 'Compat', as it depends on 'Compat/Env.hs' which defines a backward compatible type alias for 'Logger'.

I think that makes relatively sense?

Reply:
Unfortunately, there are naming collisions between `Core.hs` and `Util.hs`, and I would like to refrain from renaming functions just to avoid the naming collision.
Can we have the compromise that every module *except* for `Util` is re-exported?

Naming conflict:
https://hackage.haskell.org/package/ghc-8.10.2/docs/BasicTypes.html#t:SuccessFlag
https://hackage.haskell.org/package/ghc-8.10.2/docs/Maybes.html#t:MaybeErr

Reply:
Compat re-exports Outputable now